File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-19.cs
public class Solution {
    public int FindKthLargest(int[] nums, int k) {
        k = nums.Length - k;
        return QuickSelect(nums, 0, nums.Length - 1, k);
    }

    private int QuickSelect(int[] nums, int left, int right, int k) {
        int pivot = nums[right];
        int p = left;

        for (int i = left; i < right; i++) {
            if (nums[i] <= pivot) {
                int temp = nums[p];
                nums[p] = nums[i];
                nums[i] = temp;
                p++;
            }
        }

        int tmp = nums[p];
        nums[p] = nums[right];
        nums[right] = tmp;

        if (p > k) {
            return QuickSelect(nums, left, p - 1, k);
        } else if (p < k) {
            return QuickSelect(nums, p + 1, right, k);
        } else {
            return nums[p];
        }
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-19.cpp
class Solution {
public:
vector<vector<int>> threeSum(vector<int>& nums) {
set<vector<int>> ans;
sort(nums.begin(), nums.end());

    for (int i = 0; i < nums.size(); ++i) {
        int left = i+1;
        int right = nums.size()-1;
        while (left < right) {
            while (-nums[i] < nums[left]+nums[right]) --right;
            if (nums[left]+nums[right] == -nums[i]) {
                vector<int> triple{nums[i], nums[left], nums[right]};
                sort(triple.begin(), triple.end());
                ans.insert(triple);
            }
            ++left;
        }
    }
    return vector<vector<int>>(ans.begin(), ans.end());
}
};